sensors/src/main/scala/com/lynbrookrobotics/potassium/sensors/imu/DigitalGyro.scala

```diff
@@ -0,0 +1,75 @@
+package com.lynbrookrobotics.potassium.sensors.imu
+
+import java.util
+
+import com.lynbrookrobotics.potassium.{PeriodicSignal, Signal}
+import squants.motion.Velocity
+
+abstract class DigitalGyro {
+  // Tick Period of the robot
+  final var TICK_PERIOD: Double = ??? //TODO set value based on final 2017 robot
+  val time: squants.Time = squants.Seconds(TICK_PERIOD / 1000)
+  val length: squants.Length = squants.Meters(0)
+  val velocityZero: Velocity = Velocity(length,time)
+  var currentVelocity: Value3D[Velocity] = Value3D[Velocity](velocityZero, velocityZero, velocityZero)
+  var currentPosition: Value3D[Double] = Value3D[Double](0, 0, 0)
+  var currentDrift: Value3D[Double] = null
+
+  // List of velocities used for calibration of IMU
+  var calibrationVelocities: util.ArrayList[Value3D[Double]] = new util.ArrayList(200)
+
+  // Current index in calibrationVeocities
+  var index: Int = 0
+
+  // Whether IMU is calibrating
+  var calibrating: Boolean = true
+
+  // Gets the current velocity.
+  def retrieveVelocity: Value3D[Double]
+
+  // Update velocity stored calibrationVelocities at index
+  def calibrateUpdate: Unit = {
+    currentVelocity = retrieveVelocity
+    calibrationVelocities.set(index, currentVelocity)
+    index += 1
+
+    if (index >= 200) {
+      index = 0
+    }
+  }
+
+  // Updates values for the angle on the gyro.
+  def angleUpdate: Unit = {
+    if (calibrating) {
+      val sum: Value3D[Double] = Value3D[Double](0, 0, 0)
+      calibrationVelocities.forEach(
+        v => sum + v
+      )
+
+      currentDrift = sum.times(-1D / calibrationVelocities.size())
+      calibrationVelocities = null
+      calibrating = false
+    }
+
+    // Stores the value as a form of memory
+    // Modifies velocity according to drift and change in position
+    currentVelocity = retrieveVelocity + currentDrift
+    currentPosition += currentVelocity.times(TICK_PERIOD)
+  }
+
+  val velocity: PeriodicSignal[Value3D[Double]] = Signal {
+    if (calibrating) {
+      Value3D[Double](0, 0, 0)
+    } else {
+      retrieveVelocity
```

Comment:
This should make a call to `angleUpdate` and return `currentVelocity`, because we need to have drift correction here.